include/up-cpp/client/Notifier.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_CLIENT_NOTIFIER_H
+#define UP_CPP_CLIENT_NOTIFIER_H
+
+#include <uprotocol/v1/uattributes.pb.h>
+#include <uprotocol/v1/uri.pb.h>
+#include <uprotocol/v1/ustatus.pb.h>
+#include <up-cpp/datamodel/builder/UMessage.h>
+#include <up-cpp/transport/UTransport.h>
+
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <tuple>
+#include <utility>
+
+namespace uprotocol::client {
+
+/// @brief Interface for uEntities to send a notification to a specified target
+///
+/// Like all L2 client APIs, the Notifier is a wrapper on top of the L1
+/// UTransport API; in this instance, it provides for the notification model.
+struct Notifier {
```

Comment:
Any reason for preferring struct over class ?

Reply:
The main reason is that almost every class is written like so:
```
class Foo {
public:
    ...
private:
    ...
};
```

This is exactly the same as writing:
```
struct Foo {
    ...
private:
    ...
};
```

It removes the redundant access specifier and makes it clear that _most_ of this object is intended to be public. There are very few times that anyone would want to put private members at the top of their object declaration, too. It is cleaner to use `struct` in these scenarios and, in my opinion, helps us get away from the view that class and struct are somehow different things. They are not different in C++ beyond the default access type.